### PR TITLE
feat: use angular schematic to generate component

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change

Use angular schematic to generate component

### CODE REVIEW

File: packages/@o3r/core/schematics/component/container/index.ts [hunk 1]
```diff
@@ -70,45 +69,175 @@ export function ngGenerateComponentContainer(options: NgGenerateComponentContain
...
+        skipTests: false,
+        standalone: options.standalone,
+        skipImport: true,
+        flat: true
       }),
-      renameTemplateFiles(),
-      move(componentDestination)
-    ]), MergeStrategy.Overwrite));
+      // Angular schematics generate spec file with this pattern: component-name.component.spec.ts
+      move(componentPath.replace(/.ts$/, '.spec.ts'), specPath),
```
Comment (mrednic-1A) [inline packages/@o3r/core/schematics/component/container/index.ts:100]: it is not `replace(/component.spec.ts/, '.spec.ts'`) ?

File: packages/@o3r/core/schematics/index.it.spec.ts [hunk 2]
```diff
@@ -154,7 +154,7 @@ describe('new Otter application', () => {
     addImportToAppModule('TestAddConfigComponentContModule', 'src/components/test-add-config-component');
 
     execSync(`yarn ng g @o3r/core:component --defaults=true test-add-theming-component ${defaultOptions}`, execAppOptions);
-    execSync('yarn ng g @o3r/styling:add-theming --defaults=true --path="src/components/test-add-theming-component/presenter/test-add-theming-component-pres.style.scss"', execAppOptions);
+    execSync('yarn ng g @o3r/styling:add-theming --defaults=true --path="src/components/test-add-theming-component/presenter/test-add-theming-component-pres.component.scss"', execAppOptions);
```
Comment (mrednic-1A) [inline packages/@o3r/core/schematics/index.it.spec.ts:157]: It's the ng generator which is adding this suffix ? Should we do a `move` to rename the file and keep it with ``.style``, the same you've done for `spec`

File: packages/@o3r/core/schematics/rule-factories/otter-environment/index.ts [hunk 3]
```diff
@@ -93,9 +91,8 @@ export function updateOtterEnvironmentAdapter(
       });
 
     }
-    if (options.isDefaultGenerator) {
-      registerCollectionSchematics(workspace, '@o3r/core');
-    }
+    registerCollectionSchematics(workspace, '@schematics/angular');
+    registerCollectionSchematics(workspace, '@o3r/core');
```
Comment (mrednic-1A) [inline packages/@o3r/core/schematics/rule-factories/otter-environment/index.ts:95]: Should not be the `o3r/core` first ? and then the schematics/angular ?

File: packages/@o3r/localization/schematics/localization-to-component/index.ts [hunk 4]
```diff
@@ -38,8 +38,13 @@ const localizationProperties = [
 
 const checkLocalization = (componentPath: string, tree: Tree, baseFileName: string) => {
   const files = [
+<<<<<<< HEAD
```
Comment (mrednic-1A) [inline packages/@o3r/localization/schematics/localization-to-component/index.ts:41]: :D

File: packages/@o3r/core/schematics/rule-factories/otter-environment/index.ts (hunk 3 above)
Comment (matthieu-crouzet) [reply packages/@o3r/core/schematics/rule-factories/otter-environment/index.ts:95]: In fact no need to add `@schematics/angular` it's already done in `registerCollectionSchematics`
and `registerCollectionSchematics` is adding the collection on top of the previous registered

Comment (matthieu-crouzet) [reply packages/@o3r/core/schematics/rule-factories/otter-environment/index.ts:95]: In fact no need to add @schematics/angular it's already done in registerCollectionSchematics
and registerCollectionSchematics is adding the collection on top of the previous registered

File: packages/@o3r/localization/schematics/localization-to-component/index.ts (hunk 4 above)
Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/localization-to-component/index.ts:41]: oups :)

**Review by mrednic-1A — DISMISSED** [review]

File: packages/@o3r/core/schematics/component/container/index.ts [hunk 5]
```diff
@@ -70,45 +72,226 @@ export function ngGenerateComponentContainer(options: NgGenerateComponentContain
...
+            modulePath,
+            t.readText(modulePath),
+            ts.ScriptTarget.ES2020,
+            true
+          );
+          const changes = addExportToModule(moduleSourceFile, modulePath, `${properties.componentName}Component`, `./${properties.name}.component`);
+          const recorder = t.beginUpdate(modulePath);
+          applyToUpdateRecorder(recorder, changes);
+          t.commitUpdate(recorder);
+          return t;
```
Comment (fpaul-1A) [inline packages/@o3r/core/schematics/component/container/index.ts:162]: couldn't you use the `--module` option of the component generator to do that ?

Comment (matthieu-crouzet) [reply packages/@o3r/core/schematics/component/container/index.ts:162]: the `--module` is to expose the component in an existing module 
it's not working if the module doesn't exist

**Review by fpaul-1A — DISMISSED** [review]

**Review by fpaul-1A — DISMISSED** [review]

**Review by kpanot — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Pull request merged**